src/util.js

```diff
@@ -0,0 +1,200 @@
+ */
+export function getCourses(lessonContext, iconContext) {
+  const paths = lessonContext.keys();
+
+  return paths.reduce((res, path) => {
+    const courseName = path.slice(2, path.indexOf('/', 2));
+    const lessonFrontMatter = lessonContext(path).frontmatter;
+    const tags = cleanseTags(lessonFrontMatter.tags);
+    const lesson = {name: lessonFrontMatter.title, tags: tags};
+    const index = res.findIndex(course => course.name === courseName);
+
+    // If course already exists, push the new lesson. Else make a new course
+    if(index >= 0) {
+      res[index].lessons.push(lesson);
+    }else{
+      res.push({
+        lessons: [lesson],
+        name: courseName,
+        iconPath: iconContext('./' + courseName + '/logo-black.png'),
+        path: path
+      });
+    }
+
+    return res;
+  }, []);
+}
+
+/**
+ * Will combine lists in same properties from objA and objB. The lists will be combined as a set of distinct elements
+ * @param {Object} objA
+ * @param {Object} objB
+ * @returns {Object} merged object
+ */
+export function mergeObjects(objA, objB) {
+  const allProps = [...new Set(Object.keys(objA).concat(Object.keys(objB)))];
+  return allProps.reduce((res, propName) => {
+    const listItemsA = objA.hasOwnProperty(propName) ? objA[propName] : [];
+    const listItemsB = objB.hasOwnProperty(propName) ? objB[propName] : [];
+    res[propName] = [...new Set(listItemsA.concat(listItemsB))];
+    return res;
+  }, {});
+}
+
+/**
+ * Add tag to filter
+ * @param {Object} tag
+ * @param {Object} filter
+ * @returns {Object} filter with tag merged into it
+ */
+export function addTagToFilter(tag, filter) {
+  return mergeObjects(filter, tag);
+}
+
+/**
+ * Removes tag from filter
+ * @param {Object} tag
+ * @param {Object} filter
+ * @returns {Object} filter without the tag
+ */
+export function removeTagFromFilter(tag, filter) {
+  // Check if tag is empty or contains more than one tagGroup
+  if(Object.keys(tag).length !== 1) return filter;
+  
+  // Tag has one tagGroup with one tagItem
+  const groupName = Object.keys(tag)[0];
+  const tagName = tag[groupName].toString();
+  
+  // Check if tag contains none or more than one tagItem
+  if(tag[groupName].length !== 1) return filter;
+
+  return Object.keys(filter).reduce((res, filterGroupName) => {
+    const filterTagItems = filter[filterGroupName];
+    // Add all items except the one to be removed
+    const itemsToAdd = filterTagItems.filter(filterTagName => !(filterGroupName === groupName && filterTagName === tagName));
+    if(itemsToAdd.length === 0) return res;
+    return mergeObjects(res, {[filterGroupName]: itemsToAdd});
+  }, {});
+}
+
+/**
+ * Get courses that has at least one lesson that matches filter
+ * @param {Array} courses
+ * @param {Object} filter
+ * @returns {Array} filtered courses
+ */
+export function filterCourses(courses, filter) {
+  const coursesWithFilteredLessons = courses.map(course => {
+    const newCourse = {...course}
```

Comment:
Mangler semicolon.
Kjør gjerne eslint før du committer, så finner den sånt for deg:
```
npm run eslint -s
```